includes/api/class-wc-rest-product-variations-controller.php

```diff
@@ -248,11 +249,17 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 
 		$variation->set_parent_id( absint( $request['product_id'] ) );
 
-		// Status.
+		// TODO Fix/Remove in future API version bump. This controls the wrong thing.
+		// https://github.com/woocommerce/woocommerce/issues/15215
 		if ( isset( $request['visible'] ) ) {
 			$variation->set_status( false === $request['visible'] ? 'private' : 'publish' );
 		}
 
+		// Status.
```

Comment:
This creates a difficult set of dependencies.
Like:

> `status` available since REST API v2 and starting only after WooCommerce 3.1.0

So instead of only check for REST API version you need to check the WooCommerce version too.